tests/integration-tests/clusters_factory.py

```diff
@@ -199,6 +205,42 @@ def cfn_resources(self):
             self.__cfn_resources = retrieve_cfn_resources(self.cfn_name, self.region)
         return self.__cfn_resources
 
+    @property
+    def head_node_substack_cfn_resources(self):
+        """
+        Return the CloudFormation stack resources for the cluster's head node substack.
+        Resources are retrieved only once and then cached.
+        """
+        if not self.__head_node_substack_cfn_resources:
+            self.__head_node_substack_cfn_resources = retrieve_cfn_resources(
+                self.cfn_resources.get("MasterServerSubstack"), self.region
+            )
+        return self.__head_node_substack_cfn_resources
+
+    @property
+    def compute_fleet_substack_cfn_resources(self):
+        """
+        Return the CloudFormation stack resources for the cluster's compute fleet substack.
+        Resources are retrieved only once and then cached.
+        """
+        if not self.__compute_fleet_substack_cfn_resources:
+            self.__compute_fleet_substack_cfn_resources = retrieve_cfn_resources(
+                self.cfn_resources.get("ComputeFleetHITSubstack"), self.region
+            )
+        return self.__compute_fleet_substack_cfn_resources
+
+    @property
+    def ebs_substack_cfn_resources(self):
+        """
+        Return the CloudFormation stack resources for the cluster's EBS substack.
+        Resources are retrieved only once and then cached.
+        """
+        if not self.__compute_fleet_substack_cfn_resources:
```

Comment:
you forgot to rename `__compute_fleet_substack_cfn_resources`

Reply:
oof